moped-editor/src/components/GridTable/GridTableExport.js

```diff
@@ -170,26 +178,28 @@ const GridTableExport = ({ query, showFilterState }) => {
   };
 
   /**
-   * Update the export whenever the dialog is open
+   * Update the export whenever limit or selectall change
    */
   useEffect(() => {
     if (dialogOpen) {
+      query.limit = 0;
       getExport();
-    } else {
-      // Reset the query limit
-      query.limit = initialLimit;
     }
-  }, [dialogOpen, getExport, initialLimit, query.limit]);
+  }, [dialogOpen, query.limit, getExport]);
 
   /**
-   * Update the export whenever limit or selectall change
+   * Make the data download
    */
-  useEffect(() => {
-    if (dialogOpen) {
-      query.limit = limit;
-      getExport();
-    }
-  }, [limit, selectAll, dialogOpen, query.limit, getExport]);
+  if (dialogOpen && !loading && data) {
+    const formattedData = formatExportData(data[query.table]);
+    const headers = dataGetHeaders(formattedData);
+    const csvString = dataToCSV(headers, formattedData);
+    setTimeout(() => {
+      // Update the state
+      setDialogOpen(false);
+      downloadFile(csvString);
```

Comment:
To track how many times this code fired, I just added a console.log here on line 203. I'm not sure exactly how it is happening, but maybe wrapping this top level code in a useEffect could help control when this fires?

<img width="294" alt="Screen Shot 2021-01-25 at 11 22 48 AM" src="https://user-images.githubusercontent.com/37249039/105741642-2d533a80-5f00-11eb-9500-17bec6b25114.png">

<img width="705" alt="Screen Shot 2021-01-25 at 11 23 43 AM" src="https://user-images.githubusercontent.com/37249039/105741669-3512df00-5f00-11eb-8362-8c6a39ea786d.png">

Reply:
Thanks @mddilley I've refactored the code so that it lives in a useEffect function. I've checked by adding a console.log to the download function and I only see one run, let me know if this fixes it.

Reply:
Awesome, I tested it out and can't replicate the multiple downloads anymore. 👍